### PR TITLE
`usePlacementAliases` refactor and `useDeprecated` bugfix

### PR DESCRIPTION
**Pull request by aluarius**
## Description
- [x] `useDeprecated` composable fix (not proper worked with props).
- [x] `usePlacementAliases` (mostly types).

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/composables/usePlacementAliases.ts [hunk 1]
```diff
@@ -1,18 +1,37 @@
-export type PlacementPosition = 'top' | 'bottom' | 'left' | 'right'
+type YPlacementPosition = 'top' | 'bottom'
+type XPlacementPosition = 'left' | 'right'
+type PlacementPosition = XPlacementPosition | YPlacementPosition
+type PlacementPositionWithDefault = PlacementPosition | 'auto'
+type PlacementAlias = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right' | 'left-top' | 'right-top' | 'left-bottom' | 'right-bottom'
```
Comment (rustem-nasyrov) [inline packages/ui/src/composables/usePlacementAliases.ts:5]: Isn't possible to replace this like in #8 line below?

File: packages/ui/src/composables/usePlacementAliases.ts [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
-  .reduce((acc, position) => [...acc, position, `${position}-start`, `${position}-end`, `${position}-center`], ['auto'] as string[])
-
-export const placementAliasesPositions = ['top', 'bottom']
-  .flatMap(position => [`${position}-left`, `${position}-right`, `left-${position}`, `right-${position}`])
-
-export const placementsPositionsWithAliases = [...placementsPositions, ...placementAliasesPositions]
-
-export const aliasToPlacement: Record<string, Placement> = {
+const yPlacementPosition: YPlacementPosition[] = ['top', 'bottom']
+const xPlacementPosition: XPlacementPosition[] = ['left', 'right']
```
Comment (rustem-nasyrov) [inline packages/ui/src/composables/usePlacementAliases.ts:12]: Can be named as `verticalPlacement`/`horizontalPlacement`. But it's for your discretion.

**Review by rustem-nasyrov — CHANGES_REQUESTED** [review]
Facing issues with auto placement on the scroll. You can check it on the date-picker page for example.

**Comment by rustem-nasyrov** [comment]
I don't know is it problem with dropdown or time picker component configuration, but I'm facing this bug. When you open focus on time picker component and scroll up, it automatically positionates to the center.
Before scrolling:
image
After:
image

**Comment by aluarius** [comment]
@rustem-nasyrov this behaviour is not related to this pr, because our prod site also contains it. But yeah, it looks like a bug, will move it to the separate issue.

**Review by rustem-nasyrov — APPROVED** [review]

**Pull request merged**